Documentation/man/man3/DBInterface.3
.TH "DBInterface" 3 "Thu Mar 22 2018" "SRec Walker Counter" \" -*- nroff -*-
.ad l
.nh
.SH NAME
DBInterface \- 
.SH SYNOPSIS
.br
.PP
.PP
Inherited by \fBDBAPI\fP\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fB__construct\fP ($db)"
.br
.ti -1c
.RI "\fB__destruct\fP ()"
.br
.ti -1c
.RI "\fBprintEntireDB\fP ()"
.br
.ti -1c
.RI "\fBgetTotalNumWalkers\fP ()"
.br
.ti -1c
.RI "\fBgetNumWalkersToday\fP ()"
.br
.ti -1c
.RI "\fBgetNumWalkersThisWeek\fP ()"
.br
.ti -1c
.RI "\fBgetCurrentYearTraffic\fP ()"
.br
.ti -1c
.RI "\fBgetTrafficByYear\fP ($year)"
.br
.ti -1c
.RI "\fBgetCurrentMonthTraffic\fP ()"
.br
.ti -1c
.RI "\fBgetTrafficByMonth\fP ($year, $month)"
.br
.ti -1c
.RI "\fBgetCurrentDayTraffic\fP ()"
.br
.ti -1c
.RI "\fBgetTrafficByDay\fP ($year, $month, $day)"
.br
.ti -1c
.RI "\fBgetTrafficTimeRange\fP ($year1, $month1, $day1, $year2, $month2, $day2)"
.br
.in -1c
.SH "Detailed Description"
.PP 
Created by PhpStorm\&. User: JonathanWesterfield Date: 2/8/18 Time: 4:36 PM 
.PP
Definition at line 11 of file DBInterface\&.php\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "__construct ( $db)"
\fBDBAPI\fP constructor\&. 
.PP
\fBParameters:\fP
.RS 4
\fI$db\fP Mostly sets up the dates in this object\&. Also sets the timezone to our timezone\&. 
.RE
.PP

.PP
Implemented in \fBDBAPI\fP\&.
.SS "__destruct ()"

.PP
Implemented in \fBDBAPI\fP\&.
.SH "Member Function Documentation"
.PP 
.SS "getCurrentDayTraffic ()"

.PP
\fBReturns:\fP
.RS 4
array
.RE
.PP
Returns a length 24 array containing total number of walkers for each hour for the current day 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getCurrentMonthTraffic ()"

.PP
\fBReturns:\fP
.RS 4
array
.RE
.PP
Returns an array containing total number of walkers for each day for the current month 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getCurrentYearTraffic ()"

.PP
\fBReturns:\fP
.RS 4
array
.RE
.PP
Returns an array containing total number of walkers for each month for the current year 12 Element Array 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getNumWalkersThisWeek ()"

.PP
\fBReturns:\fP
.RS 4
int
.RE
.PP
Returns the number of walkers from the start to the end of the week 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getNumWalkersToday ()"

.PP
\fBReturns:\fP
.RS 4
int
.RE
.PP
Returns the number of walkers that walked through from the start to the end of the day 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getTotalNumWalkers ()"

.PP
\fBReturns:\fP
.RS 4
int
.RE
.PP
Returns the total number of walkers in the table 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getTrafficByDay ( $year,  $month,  $day)"

.PP
\fBParameters:\fP
.RS 4
\fI$year\fP 
.br
\fI$month\fP 
.br
\fI$day\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
array
.RE
.PP
Returns an array (24) containing total number of walkers for each hour
.PP
Usage: <var = getTrafficByDay(2018, 2, 15);> for Febraury 15, 2018 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getTrafficByMonth ( $year,  $month)"

.PP
\fBParameters:\fP
.RS 4
\fI$year\fP 
.br
\fI$month\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
array
.RE
.PP
Gets the traffic for each day during the specified month of the specified year
.PP
Usage: getTrafficByMonth(2018, 2); // for February 2018 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getTrafficByYear ( $year)"

.PP
\fBParameters:\fP
.RS 4
\fI$year\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
array
.RE
.PP
Gives the traffic for each month in an array for the specified year passed in 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "getTrafficTimeRange ( $year1,  $month1,  $day1,  $year2,  $month2,  $day2)"

.PP
\fBParameters:\fP
.RS 4
\fI$year1\fP 
.br
\fI$month1\fP 
.br
\fI$day1\fP 
.br
\fI$year2\fP 
.br
\fI$month2\fP 
.br
\fI$day2\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
int
.RE
.PP
Takes in a date range (start and end date) and counts the number of walkers in the given range 
.PP
Implemented in \fBDBAPI\fP\&.
.SS "printEntireDB ()"

.PP
Implemented in \fBDBAPI\fP\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for SRec Walker Counter from the source code\&.


DBInterface.php
<?php
/**
 * Created by PhpStorm.
 * User: JonathanWesterfield
 * Date: 2/8/18
 * Time: 4:36 PM
 */

include_once "DBAPI.php";

interface DBInterface
{
    /**
     * DBAPI constructor.
     * @param $db
     *
     * Mostly sets up the dates in this object. Also sets the timezone to our timezone.
     */
    public function __construct($db);

    // class Destructor
    public function __destruct();

    // prints out the entire database on the page in a table
    public function printEntireDB();

    /**
     * @return int
     *
     * Returns the total number of walkers in the table
     */
    public function getTotalNumWalkers();

    /**
     * @return int
     *
     * Returns the number of walkers that walked through from the start to the end of the day
     */
    public function getNumWalkersToday();

    /**
     * @return int
     *
     * Returns the number of walkers from the start to the end of the week
     */
    public function getNumWalkersThisWeek();

    //public function getHotTimes(); // gets the hours of day with peak traffic
    //public function getHotDays(); // gets the days with peak traffic for a week

    /**
     * @return array
     *
     * Returns an array containing total number of walkers for each month for the current year
     * 12 Element Array
     */
    public function getCurrentYearTraffic();

    /**
     * @param $year
     * @return array
     *
     * Gives the traffic for each month in an array for the specified year passed in
     */
    public function getTrafficByYear($year);

    /**
     * @return array
     *
     * Returns an array containing total number of walkers for each day for the current month
     */
    public function getCurrentMonthTraffic();

    /**
     * @param $year
     * @param $month
     * @return array
     *
     * Gets the traffic for each day during the specified month of the specified year
     *
     * Usage: getTrafficByMonth(2018, 2); // for February 2018
     */
    public function getTrafficByMonth($year, $month);

    /**
     * @return array
     *
     * Returns a length 24 array containing total number of walkers for each hour
     * for the current day
     */
    public function getCurrentDayTraffic();

    /**
     * @param $year
     * @param $month
     * @param $day
     * @return array
     *
     * Returns an array (24) containing total number of walkers for each hour
     *
     * Usage: <var = getTrafficByDay(2018, 2, 15);> for Febraury 15, 2018
     */
    public function getTrafficByDay($year, $month, $day);

    /**
     * @param $year1
     * @param $month1
     * @param $day1
     * @param $year2
     * @param $month2
     * @param $day2
     * @return int
     *
     * Takes in a date range (start and end date) and counts the number of walkers in the given range
     */
    public function getTrafficTimeRange($year1, $month1, $day1, $year2, $month2, $day2);
}

?>